test/Makefile

```diff
@@ -0,0 +1,175 @@
+# Build the `libc-test` tests as Wasm programs and run them with the selected
+# engine. Contributors beware! This Makefile follows the style of the
+# `libc-test` Makefile and uses some more exotic features of `make`.
+#
+# The top-level `test` target is composed of a chain of several phony
+# sub-targets:
+# - `download`: retrieve the `libc-test` source from a Git `$(MIRROR)`
+# - `build`: construct Wasm modules for a subset of the `libc-test` tests
+# - `run`: execute the benchmarks with a Wasm `$(ENGINE)` of choice (e.g.,
+#   Wasmtime)
+
+test: run
+
+# Unlike the `libc-test` directory, we will output all artifacts to the `build`
+# directory (keeping with the `wasi-libc` conventions).
+OBJDIR ?= $(CURDIR)/build
+DOWNDIR ?= $(CURDIR)/download
+
+##### DOWNLOAD #################################################################
+
+LIBC_TEST_URL ?= https://github.com/bytecodealliance/libc-test
+LIBC_TEST = $(DOWNDIR)/libc-test
+LIBRT_URL ?= https://github.com/WebAssembly/wasi-sdk/releases/download/wasi-sdk-16/libclang_rt.builtins-wasm32-wasi-16.0.tar.gz
+LIBRT = $(DOWNDIR)/lib/wasi/libclang_rt.builtins-wasm32.a
+WASMTIME_URL ?= https://github.com/bytecodealliance/wasmtime/releases/download/v3.0.0/wasmtime-v3.0.0-x86_64-linux.tar.xz
+WASMTIME = $(DOWNDIR)/$(shell basename $(WASMTIME_URL) .tar.xz)/wasmtime
+
+download: $(LIBC_TEST) $(LIBRT) $(WASMTIME)
+
+$(DOWNDIR):
+	mkdir -p download
+
+$(LIBC_TEST): | $(DOWNDIR)
+	git clone --depth 1 $(LIBC_TEST_URL) $@
```

Comment:
Why not use a git submodule instead?

Reply:
There were some concerns about licensing; some parts (unused here) are [licensed under GPL](http://nsz.repo.hu/git/?p=libc-test;a=blob;f=COPYRIGHT;h=67acf495aa00c48cf1b9334b12a94a83a28b6b8a;hb=HEAD). In discussions with @sunfishcode, we felt that downloading the code on demand and never releasing any of the libc-test parts avoided license contamination with this repo. What do you think?

Reply:
I don't see how doing a `git clone` programmatically is any different to using a git submodule, but IANAL..

Reply:
Yeah, I see what you mean; I guess I had thought that cloning it at a different time made it more optional (i.e., only those who want to run tests). I set up a meeting to talk to someone here about it because... IAANAL.